templates/etc/ferm/rules.d/rule.conf.j2

```diff
@@ -0,0 +1,626 @@
+{%     elif config.ports|d() %}
+{%       set _ = ferm__tpl_config.update({'dmz_ports': ([ config.ports ] if config.ports is string else config.ports) }) %}
+{%     endif %}
+{%   endif %}
+{%   if config.interface_present|d() %}
+{%     if config.interface_present is string %}
+{%       if hostvars[inventory_hostname]["ansible_" + config.interface_present]|d() %}
+{%         set _ = ferm__tpl_config.update({'interface_present': [ config.interface_present ] }) %}
+{%       endif %}
+{%     else %}
+{%       for interface in config.interface_present %}
+{%         if hostvars[inventory_hostname]["ansible_" + interface]|d() %}
+{%           set _ = ferm__tpl_config.update({'interface_present': [ interface ] }) %}
+{%         endif %}
+{%       endfor %}
+{%     endif %}
+{%   elif config.interfaces_present|d() %}
+{%     if config.interfaces_present is string %}
+{%       if hostvars[inventory_hostname]["ansible_" + config.interfaces_present]|d() %}
+{%         set _ = ferm__tpl_config.update({'interface_present': [ config.interface_present ] }) %}
+{%       endif %}
+{%     else %}
+{%       for interface in config.interfaces_present %}
+{%         if hostvars[inventory_hostname]["ansible_" + interface]|d() %}
+{%           set _ = ferm__tpl_config.update({'interface_present': [ interface ] }) %}
+{%         endif %}
+{%       endfor %}
+{%     endif %}
+{%   endif %}
+{%   if config.outerface|d() %}
+{%     set _ = ferm__tpl_config.update({'outerface': ([ config.outerface ] if config.outerface is string else config.outerface) }) %}
+{%   elif config.outerfaces|d() %}
+{%     set _ = ferm__tpl_config.update({'outerface': ([ config.outerfaces ] if config.outerfaces is string else config.outerfaces) }) %}
+{%   endif %}
+{%   if config.outerface_present|d() %}
+{%     if config.outerface_present is string %}
+{%       if hostvars[inventory_hostname]["ansible_" + config.outerface_present]|d() %}
+{%         set _ = ferm__tpl_config.update({'outerface_present': [ config.outerface_present ] }) %}
+{%       endif %}
+{%     else %}
+{%       for outerface in config.outerface_present %}
+{%         if hostvars[inventory_hostname]["ansible_" + outerface]|d() %}
+{%           set _ = ferm__tpl_config.update({'outerface_present': [ outerface ] }) %}
+{%         endif %}
+{%       endfor %}
+{%     endif %}
+{%   elif config.outerfaces_present|d() %}
+{%     if config.outerfaces_present is string %}
+{%       if hostvars[inventory_hostname]["ansible_" + config.outerfaces_present]|d() %}
+{%         set _ = ferm__tpl_config.update({'outerface_present': [ config.outerfaces_present ] }) %}
+{%       endif %}
+{%     else %}
+{%       for outerface in config.outerfaces_present %}
+{%         if hostvars[inventory_hostname]["ansible_" + outerface]|d() %}
+{%           set _ = ferm__tpl_config.update({'outerface_present': [ outerface ] }) %}
+{%         endif %}
+{%       endfor %}
+{%     endif %}
+{%   endif %}
+{%   if config.protocol|d() %}
```

Comment:
To here: )

This can be expressed in like 4 lines of code with the help of debops macros :wink: